lib/realtime/tenants/scheduled_message_cleanup.ex

```diff
@@ -54,17 +62,27 @@ defmodule Realtime.Tenants.ScheduledMessageCleanup do
   @impl true
   def handle_info(:delete_old_messages, state) do
     Logger.info("ScheduledMessageCleanup started")
-    %{region: region, chunks: chunks} = state
+    %{region: region, chunks: chunks, task_timeout: task_timeout} = state
     regions = Nodes.region_to_tenant_regions(region)
     region_nodes = Nodes.region_nodes(region)
 
     Realtime.Repo.transaction(fn ->
-      from(t in Tenant, select: t)
+      from(t in Tenant,
+        join: e in assoc(t, :extensions),
+        preload: :extensions,
+        where: t.notify_private_alpha == true
+      )
       |> where_region(regions)
-      |> Repo.stream()
+      |> Repo.all()
       |> Stream.filter(&node_responsible_for_cleanup?(&1, region_nodes))
       |> Stream.chunk_every(chunks)
-      |> Enum.map(&run_cleanup_on_tenants/1)
+      |> Enum.map(fn chunks ->
+        Task.Supervisor.async(
```

Comment:
> Raises an error if supervisor has reached the maximum number of children.

This GenServer would crash then if you reach max_children. 

Which might be fine, but I would expect you to reach max_children potentially frequently once we're doing this for a lot of dbs. 

Consider using `Task.Supervisor.start_child` to handle the max_children error tuple.

Reply:
went with async_nolink and set a limit at the supervisor level